Allocations: use double check icon and add status colors

e18r opened the pull request:
Fixes #1763 

* Moved the double check icon from Rewards to a shared location
* Updated the import from Rewards and tested it
* Replaced the check icon with the double check icon in Allocations
* Added status colors
* Discovered a bug whereby only allocations with enacted status show up in Allocations History; created ticket #1852 to track it

![Screenshot from 2020-01-08 12-16-10](https://user-images.githubusercontent.com/16065447/72001596-c6d77d80-3213-11ea-8ec8-fbf5748abf7a.png)

![Screenshot from 2020-01-08 12-31-55](https://user-images.githubusercontent.com/16065447/72001603-ca6b0480-3213-11ea-850c-3302a983adfc.png)
(this last one was updated by manually changing the status)

e18r committed:
fix(allocations): use double check icon and add status colors

Moved the double check icon from Rewards to a shared location; updated the import from Rewards and tested it; replaced the check icon with the double check icon in Allocations; added status colors; discovered a bug whereby only Allocations with enacted status show up in Allocations History, created ticket #1852 to track it.

chadoh commented on apps/allocations/app/components/App/AllocationsHistory.js:
```diff
@@ -18,6 +19,8 @@ import { STATUSES } from '../../utils/constants'
 import { displayCurrency } from '../../../../../shared/ui/helpers'
 import { addressesEqual } from '../../../../../shared/lib/web3-utils'
 
+const IconDoubleCheck = () => <img src={doubleCheck} alt="" />
```
Let's export a usable component from the shared location, so the interface stays similar to the one used by aragonUI

Also, I just want to think aloud about whether `alt=""` is the correct option. I think it probably is, here, since it doesn't add anything to the experience if a screen reader reads something aloud like "double checkmark enacted" instead of just "enacted." Do you agree?

Note that I think in most places we are using inline `svg` tags, which seems better to me. That gives us more control over accessibility, rather than just the `alt` text ([read more](https://www.sitepoint.com/tips-accessible-svg/)), and it allows us to override styles such as the icon color easily using CSS.

chadoh commented on apps/allocations/app/components/App/AllocationsHistory.js:
```diff
@@ -160,8 +163,11 @@ const RecipientAmount = styled.div`
-    theme.contentSecondary : theme.content};
+  color: ${({ code, theme }) => code === 0
+    ? theme.contentSecondary
+    : code === 1
```
Yikes, this `code` argument is really hard to understand – it's _probably_ out-of-scope to improve this now, but I would love to see this variable or its values get more-understandable names.

chadoh commented on apps/rewards/app/utils/helpers.js:
```diff
@@ -4,7 +4,8 @@ import {
 } from './constants'
-import doubleCheck from '../assets/double-check.svg'
+import doubleCheck
+  from '../../../../shared/ui/components/assets/svg/double-check.svg'
```
Hmm, I see why we stuck with exporting the raw `svg` rather than a component – it's what we were already doing in Rewards. It looks like this is the only file that uses it, though, so it would be quick to switch to a component instead of just a raw svg file.

Also, it will give us a chance to switch to exporting an inline `<svg>` rather than an `<img>`, as [accessibility experts recommend](https://www.sitepoint.com/tips-accessible-svg/), and fix the absent `alt=` problem on the `img` in this file by adding a customizable `title` attribute to the svg.

chadoh reviewed (changes requested):
I like how small and focused this change is, and how thorough your description is! Thanks!

I'd like to see some small improvements to accessibility & code simplicity, but they shouldn't take long. See below.

e18r committed:
fix(allocations): icon element & status constants

* Following @chadoh's suggestion, I transformed the double check icon
into a component, so that it can be imported in much the same way
other icons are imported from AragonUI.

* I also updated the way we represent the status of each allocation in
Allocations History. Now, instead of numbers, a more explanatory
string constant is used.

e18r commented:
Hi @chadoh, thanks for your review. I addressed both issues you mentioned, so now I consider this PR to be ready for another round of review. Thank you!

chadoh reviewed (approved):
👏
